### PR TITLE
[Builtins] Develop the parsing of polymorphic data PR

### PR DESCRIPTION
**Pull request by gtrepta**

### CODE REVIEW

File: plutus-core/plutus-core/src/PlutusCore/Parser.hs [hunk 1]
```diff
@@ -101,18 +101,3 @@ parseTerm = parseGen term
...
--- >>> doIt term "(con bool True)"
--- Right (Constant (SourcePos {sourceName = "test", sourceLine = Pos 1, sourceColumn = Pos 2}) (Some (ValueOf bool True)))
--- >>> doIt term "(con (list bool) [True, False])"
--- Right (Constant (SourcePos {sourceName = "test", sourceLine = Pos 1, sourceColumn = Pos 2}) (Some (ValueOf list (bool) [True,False])))
--- >>> doIt term "(con (pair unit (list integer)) ((),[1,2,3]))"
--- Right (Constant (SourcePos {sourceName = "test", sourceLine = Pos 1, sourceColumn = Pos 2}) (Some (ValueOf pair (unit) (list (integer)) ((),[1,2,3]))))
--- >>> doIt term "(con (list (pair string bool)) [(\"abc\", True), (\"def\", False)])"
--- Right (Constant (SourcePos {sourceName = "test", sourceLine = Pos 1, sourceColumn = Pos 2}) (Some (ValueOf list (pair (string) (bool)) [("abc",True),("def",False)])))
--- >>> doIt term "(con string \"abc\")"
--- Right (Constant (SourcePos {sourceName = "test", sourceLine = Pos 1, sourceColumn = Pos 2}) (Some (ValueOf string "abc")))
```
Comment (effectfully) [inline plutus-core/plutus-core/src/PlutusCore/Parser.hs:118]: Could you create unit tests out of these? They are helpful examples.

File: plutus-core/plutus-core/src/PlutusCore/Parser/Type.hs [hunk 2]
```diff
@@ -87,25 +83,3 @@ defaultUniType = do
...
--- >>> doIt defaultUniType "list integer"
--- Right (SomeTypeIn (Kinded list (integer)))
--- >>> doIt defaultUniType "pair (list bool)"
--- Right (SomeTypeIn (Kinded pair (list (bool))))
--- >>> doIt defaultUniType "pair (list unit) integer"
--- Right (SomeTypeIn (Kinded pair (list (unit)) (integer)))
--- >>> doIt defaultUniType "list (pair unit integer)"
--- Right (SomeTypeIn (Kinded list (pair (unit) (integer))))
--- >>> doIt defaultUniType "pair unit (pair bool integer)"
--- Right (SomeTypeIn (Kinded pair (unit) (pair (bool) (integer))))
```
Comment (effectfully) [inline plutus-core/plutus-core/src/PlutusCore/Parser/Type.hs:111]: Ditto.

**Review by effectfully — COMMENTED** [review]
Thanks for the PR and welcome to the project!

It may be helpful if you check our [Contributing](https://github.com/input-output-hk/plutus/blob/master/CONTRIBUTING.adoc) guide. In particular, CI is red, because we use `stylish-haskell` for formatting various little things and CI checks that every file is formatted with this tool, which you didn't do.

Did you look into the code that does parsing of polymorphic data? Asking because it needs docs, so if you understand it, you could write the docs. Otherwise I'll do it in a separate PR myself.

File: plutus-core/plutus-core/src/PlutusCore/Parser.hs (hunk 1 above)
Comment (effectfully) [reply plutus-core/plutus-core/src/PlutusCore/Parser.hs:118]: @gtrepta FYI, I [fixed](https://github.com/input-output-hk/plutus/pull/4724) rendering of built-in types yesterday, so these old GHCi sessions are no longer correct, you'll need to regerenerate the output for each test. Also, don't use `Show` like I did, use `Pretty` instead.

**Comment by gtrepta** [comment]
> Did you look into the code that does parsing of polymorphic data? Asking because it needs docs, so if you understand it, you could write the docs. Otherwise I'll do it in a separate PR myself.

I'm afraid you will need to write the docs, I hardly understand it.

**Comment by effectfully** [comment]
Closing in favor of #4738.

**Pull request closed without merging**